### ISSUE TITLE
Unable to login via rest api - something stole my cookie!

### ISSUE THREAD

**Issue by qualityshepherd** [DevExpress/testcafe#2583]
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
I'm not able to login successfully when using rest api login, then setting cookies in the browser. The login cookie is being set (see console log within login method), but then doesn't exist inside the test for some reason (see console log within test). 

### What is the expected behavior?
Should work. I'm setting the cookie... it exists... then it doesn't. Why come?

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js
    // loginPage object
    async loginAs(userObj) {
        const response = await rest.restLogin(userObj.username, userObj.password);
        const cookie = await rest.getCookieFromRestResponse(response);
        await this.setCookie(cookie);
        // login cookie is set correctly at this point
        console.log('getting cookie: ', await this.getCookie());
        await projectsPage.goto();
    }

    async getCookie() {
        return ClientFunction(() => {
            return document.cookie;
        })();
    }

    test.only('should log in!', async t => {
        await loginPage.goto();
        await loginPage.loginAs(admin);
        // no login cookie at this point... why?
        console.log(inside test, the cookie is: ', await loginPage.getCookie());

        await t.expect(projectsPage.projectTitles.nth(0).exists).eql(true);
    });



```

### Specify your

* operating system: mac
* testcafe version: latest
* node.js version: latest lts

**Comment by miherlosev** [comment]
Hi @qualityshepherd

We cannot understand the reason of the problem without example to reproduce.
Please, create a simple example of your REST API service and provide us, if it's a possible. 
Or just provide a link to your application if it is a public.

**Comment by LavrovArtem** [comment]
Hi @qualityshepherd,
Could you please provide parameters of the cookie you set?
Example: `key=value;Path=/some/path`.
I think your problem can be related to the `path` parameter.
The `path` parameter equals `/some/path` by default If you set the `key=value` cookie (without `path`) on the `/some/path/123` page.
This cookie is not available on the `/some` page.

**Comment by qualityshepherd** [comment]
Hi @LavrovArtem and @miherlosev,

Yeah, it looks like Testcafe is deleting the session cookie when I try to navigate from my login page (`/signin`), to my app's landing page (`/accounts/`). Is that expected, and if so, what can I do to work around this?

The cookie I'm setting is: `id=[a valid session token]; path=/; secure; HttpOnly`

FYI, I'm porting a few of my current Protractor tests to TestCafe, to see if we might make a switch. Obviously, I need to be able to log in :) I've been over this a million times on my end and cannot see an issue from my side. Any ideas on your end?

**Comment by LavrovArtem** [comment]
Hi @qualityshepherd,
Sounds strange that you can see this cookie at the first time, because you set it with the `HttpOnly` flag. This cookie cannot be available on the client side via the `document.cookie` property.

**Comment by qualityshepherd** [comment]
Yeah, I've also tried it without `HttpOnly`... same issue. I don't _think_ that's the problem... hmmmm.  

I can't give y'all access to the system, but I could jump on a quick video chat and hopefully debug this?

**Comment by AlexSkorkin** [comment]
@qualityshepherd 
 
Let's try doing this on Monday. What would be the best time for you to perform our remote session? We prefer using the TeamViewer tool. Please install it and let us know your access codes (including your appropriate time).

**Comment by qualityshepherd** [comment]
Okay, more debugging... and yeah, it's gotta be the `HttpOnly` Sadly, this might be a dealbreaker... bummer.

**Comment by qualityshepherd** [comment]
Okay... as it turns out, our app (for reasons) kicks out a cookie when you hit our signin page, that _doesn't_ let you log in, but has the same name as the login cookie, AND has the `HttpOnly` flag 😞 Thus, because Testcafe uses `document.cookie` to set cookies, it fails to overwrite. This wasn't a problem with webdriver. 

I don't suppose y'all have a workaround :)

**Comment by MarinaRukavitsyna** [comment]
@qualityshepherd 
We would be happy to help you with this issue. Please let us know if you need further assistance with this thread.

**Comment by gaborari** [comment]
I faced the same issue:
- navigating to my site
- the site navigates to login page
  (the login method uses cookies - IdentityServer)
- type email address
- then due to the missing cookies, I get unauthorized response

And I cannot see any required cookies in request

**Comment by AndreyBelym** [comment]
@gaborari, please create a [new bug report](https://github.com/DevExpress/testcafe/issues/new?template=bug-report.md) and try to provide an example that can be used to reproduce the problem.
